okhttp/src/main/java/okhttp3/OkHttpClient.kt

```diff
@@ -248,12 +248,16 @@ open class OkHttpClient internal constructor(
 
   /** Uses [request] to connect a new web socket. */
   override fun newWebSocket(request: Request, listener: WebSocketListener): WebSocket {
+    // Compress messages 1 KiB or larger.
+    val minimumDeflateSize = 1024L
     val webSocket = RealWebSocket(
-        TaskRunner.INSTANCE,
-        request,
-        listener,
-        Random(),
-        pingIntervalMillis.toLong()
+        taskRunner = TaskRunner.INSTANCE,
+        originalRequest = request,
+        listener = listener,
+        random = Random(),
+        pingIntervalMillis = pingIntervalMillis.toLong(),
+        extensions = null, // Always null for clients.
```

Comment:
RealWebSocket is an internal class, feels like these two constructor invocations could be factory methods on the companion object? random, ping and extensions both diff.

Reply:
I tried it and I ended up with clumsier code than when I started. I think the problem is that the tests are sort of clients and sort of servers, so we’d end up needing several ways to construct this.